packages/ens-referrals/src/referrer.test.ts

```diff
@@ -0,0 +1,89 @@
+import { type Address, bytesToHex, zeroAddress } from "viem";
+import { describe, expect, it } from "vitest";
+
+import { buildEncodedReferrer, buildReferrerUrl, decodeEncodedReferrer } from "./referrer";
+
+const vitalikEthAddress: Address = "0xd8da6bf26964af9d7eed9e03e53415d37aa96045";
+const vitalikEthAddressChecksummed: Address = "0xd8dA6BF26964aF9D7eEd9e03E53415D37aA96045";
+
+describe("encoded referrer", () => {
+  describe("decoding a 32-byte value", () => {
+    it("returns the EVM address when initial bytes were all zeroes", () => {
+      // arrange
+      const initialBytes = new Uint8Array([0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0]);
+
+      const remainingBytes = new Uint8Array([
+        216, 218, 107, 242, 105, 100, 175, 157, 126, 237, 158, 3, 229, 52, 21, 211, 122, 169, 96,
+        69,
+      ]);
+
+      const input = bytesToHex(new Uint8Array([...initialBytes, ...remainingBytes]));
```

Comment:
it might be clearer to construct this with `hexToBytes(pad(VITALIKS_ADDRESS, {size: 32, dir: 'left'}))` rather than doing the manual byte stuff here

https://viem.sh/docs/utilities/pad#pad

Reply:
Thanks a lot for that hint 👍